src/app/core/history.service.ts

```diff
@@ -1,40 +1,94 @@
 import { Injectable } from '@angular/core';
-import { Router, NavigationStart, NavigationEnd } from '@angular/router';
+import { Router, NavigationEnd } from '@angular/router';
 import { filter } from 'rxjs/operators';
 
+export interface HistorySnapshot {
+  /**
+   * Rewind the history of the HistoryService to the last location in the snapshot
+   *
+   * @readonly
+   * @memberof HistorySnapshot
+   */
+  rewind(): void;
+}
+
 @Injectable({
   providedIn: 'root'
 })
 export class HistoryService {
 
-  history: NavigationEnd[] = [];
+  private skipNextEvent: boolean;
+
+  history: string[] = [];
```

Comment:
Maybe adding a little JSDoc on this property would help?
```suggestion
  /**
   * A stack containing the URLs of a user's navigation history
   */
  history: string[] = [];
```